Toolkit/ArcGISToolkit/Compass.swift

```diff
@@ -67,20 +65,32 @@ public class Compass: UIImageView {
         
         // animate the compass visibility, if necessary
         animateCompass()
+        
+        // Add Compass as an observer of the mapView's rotation.
+        rotationObservation = mapView.observe(\.rotation, options: .new) { (mapView, change) in
```

Comment:
This closure needs to capture a `weak` reference to `self` in order to prevent a retain cycle.